File: implementations/js/runner.js
function runner (pageSize, numPages, numExecs) {
  var data = randCRC(numPages, pageSize)
  var numWords = pageSize / 4

  if ((pageSize % 4) !== 0) {
    console.error("Unsupported page size of '" + pageSize +
      "' please choose a page size that is a multiple of 4")
  }

  var crcs = new Uint32Array(numPages) // TA
  var cumulativeTime = 0

  var expectedCrc = 2231263667
  var t1 = performance.now()
  for (var j = 0; j < numExecs; ++j) {
    for (var i = 0; i < numPages; ++i) {
      crcs[i] = crc32_8bytes(data, i * numWords, pageSize)
    }

    // crc32_8bytes is expecting the length in bytes
    var finalCrc = crc32_8bytes(crcs, 0, numPages * 4)
  }
  cumulativeTime += performance.now() - t1

  console.log('Total time was ' + cumulativeTime / 1000 + ' s')
  console.log(JSON.stringify({ 
    status: 1,
    options: 'run(' + [numPages, pageSize, numExecs].join(',') + ')',
    time: cumulativeTime / 1000,
    output: finalCrc
  }))
}

File: implementations/matlab/runner.m
function runner(page_size,num_pages,num_execs,lut_file,input_directory)
% Example: runner(65536,128,1);
if mod(page_size, 8) ~= 0
    error('Unsupported page size of %u, please choose a page size that is a multiple of 8\n',page_size);
end

s = RandStream('mcg16807','Seed',49734321);
% RandStream.setDefaultStream(s);
RandStream.setGlobalStream(s);
seed = 10000;


expected_crc = 2231263667;

fileID = fopen(lut_file, 'r');
crc32Lookup = reshape(fscanf(fileID,'%x'),256,8)';
fclose(fileID);

file_path = fullfile(input_directory, strcat(num2str(page_size), '-', num2str(num_pages), '-data.csv'));
if exist(file_path) 
    fprintf('Loaded data from %s', file_path);
    num_words = page_size/4;
    h_num = reshape(uint32(load(file_path)), num_words, num_pages);
else
    fprintf('%s does not exist, generating data instead', file_path);
    h_num = randi(seed, num_words, num_pages); %rand_crc
end

tic
for j = 1:num_execs
    final_crc = crc_ostrich(page_size,num_pages,h_num,crc32Lookup);
end
elapsedTime = toc;

int32_crc = typecast(final_crc, 'int32');
final_crc = int32_crc(1);

fprintf('{ \"status\": %d, \"options\": \"-n %d -s %d -r %d\", \"time\": %f, \"output\": %d }\n', 1, num_pages, page_size, num_execs, elapsedTime, final_crc);

end

